### PR TITLE
Change accesses to `Weather` throughout RuFaS

### PR DESCRIPTION
**Pull request by Isgotkowitz**
This PR fully integrates `get_current_weather()` from `Weather` into the rest of the codebase, and eliminates direct accesses to `Weather`'s attributes. The files where access was modified are `animal_manager.py`, `manure_handler_classes.py`, and `base_manure_treatment.py`, and have also had type annotations added to changed methods. All associated tests are updated as well.

### CODE REVIEW

File: RUFAS/weather.py [hunk 1]
```diff
@@ -47,13 +47,13 @@ def __init__(self, weather_file, config):
...
 
         # initialize data sets
-        self.rainfall = []
-        self.T_max = []
-        self.T_min = []
-        self.T_avg = []
-        self.radiation = []
-        self.irrigation = []
-        self.T_avg_annual = []
+        self.__rainfall = []
```
Comment (JoeWaddell) [inline RUFAS/weather.py]: Would this PR be the place to improve variable names (specifically ones with 'T', like T_max), or to update the format of the Docstrings in the weather file?

**Review by JoeWaddell — COMMENTED** [review]
Nice straightforward PR. Changes make sense, and improves readability of the code overall.

**Review by JoeWaddell — APPROVED** [review]

File: RUFAS/weather.py (hunk 1 above)
Comment (Isgotkowitz) [reply RUFAS/weather.py]: It would, good thought.

**Review by ltp23 — APPROVED** [review]
Very good PR. Thanks, Ed. I like that we can now annotate the weather and time references. Thank you for updating their usage in the animal and manure module. 

One design concern I have is the use of 'private' dunder (double underscore) variables. They seem somewhat of an anti-pattern in Python because we are really going out of our way to make it inaccessible to the user. If someone wants to modify those supposedly private variables, they still can. My point is that if anything is not in the public API, it is expected that the user should not dig deeper than that.

**Comment by Isgotkowitz** [comment]
> One design concern I have is the use of 'private' dunder (double underscore) variables. They seem somewhat of an anti-pattern in Python because we are really going out of our way to make it inaccessible to the user. If someone wants to modify those supposedly private variables, they still can. My point is that if anything is not in the public API, it is expected that the user should not dig deeper than that.

@ltp23 I agree, I think those double underscore arrays are very anti-pattern like. They should be changed at some point, but after our work together session today it seems like #753 will not be closed for a while, so the private arrays might be around for a while.

**Pull request merged**